types/declarative-net-request/index.d.ts

```diff
@@ -0,0 +1,20 @@
+type DNRRule = import('../../packages/ddg2dnr/lib/utils.js').DNRRule;
+type DNRRuleWithID = import('../../packages/ddg2dnr/lib/utils.js').DNRRuleWithID;
+
+// Augment the chrome.declarativeNetRequest API type definitions as provided by
+// @types/chrome to allow for DNRRule/DNRRuleWithID to be used. This is
+// necessary since working with the original Rule type that @types/chrome
+// provides is tricky from JS code, as that uses enums for some of the argument
+// properties. Enums unfortunately aren't ergonomic to use from JavaScript code,
+// assigning a correct string value for the enum isn't enough.
+declare namespace chrome.declarativeNetRequest {
```

Comment:
Maybe @shakyShane knows but perhaps there's a way to coerce our type into theirs, seems like we aren't doing it in many places so if possible that seems preferred than overriding this which might get outdated etc.

Reply:
I couldn't see an easy way to do that. Lots of the Rule type properties are enums, whereas our DNRRuleWithID type goes with strings instead. I figured augmenting the interface for `updateDynamicRules` and `updateSessionRules` to allow our type as well was a pretty clean option for now, but open to suggestions if you can see a better way.

Reply:
I added an idea over in https://github.com/duckduckgo/duckduckgo-privacy-extension/commit/c5f2cdf0d17d18cd2d8fc65bc5e4c8c32923c951